### DISCUSSION TITLE
riscv32-unknown-elf-gcc -O3

### DISCUSSION THREAD

**Discussion by Josephood7** [bol-edu/HLS-SOC-Discussions#195] [Q&A]
之前看到有組別用 -O3 來跑run_sim, 我跑-O1 -O2燈沒問題。
但跑 -O3 的時候，LA1 PASS之後後面就錯了
不知道還有甚麼其他方要一起改的嗎

**Comment by pouree** [comment]
用O3的話，instruction code會過多，所以會導致bank超過，目前測下來應該是O2能達到最優化。

**Comment by KaiChen18** [comment]
也可以試著用Os看看 我有嘗試過可以pass

**Comment by Sheng08** [comment]
我們組別使用 -O1 -O2 -Os 都可以成功，但使用 -O3 也一樣會無法 Pass，因為 -O3 的優化有可能會 Loop Unrolling 導致 code 變多。

**Comment by boledulab** [comment]
你們可以 report code size 嗎？若是用 sdram 的， 需要從 Github lab-sdram repo 新的版本。
